proxy/src/schema.rs

```diff
@@ -209,8 +323,15 @@ impl Mutation {
         name: String,
         description: String,
         img_url: String,
-    ) -> FieldResult<Project> {
-        Ok(ctx.source.register_project(name, description, img_url))
+    ) -> Result<Project, Error> {
+        let err_name = &name.clone();
+        match ctx.source.register_project(name, description, img_url) {
```

Comment:
This pattern of keeping the `Ok` intact while changing the `Err` side is captured with `map_err` on a `Result` :)